dd-java-agent/agent-profiling/profiling-exceptions/src/main/java/com/datadog/profiling/exceptions/ExceptionProfiling.java

```
@@ -0,0 +1,82 @@
+package com.datadog.profiling.exceptions;
+
+import datadog.trace.api.Config;
+import jdk.jfr.FlightRecorder;
+import jdk.jfr.FlightRecorderListener;
+import jdk.jfr.Recording;
+import jdk.jfr.RecordingState;
+
+/**
+ * JVM-wide singleton exception profiling service. Uses {@linkplain Config} class to configure
+ * itself using either system properties, environment or properties override.
+ */
+public final class ExceptionProfiling {
+  private static volatile Config config = null;
+
+  private static final class Singleton {
+    static final ExceptionProfiling INSTANCE = new ExceptionProfiling();
+  }
+
+  /**
+   * Initialization routine. !!! MUST BE CALLED BEFORE {@linkplain ExceptionProfiling#getInstance()}
+   * - otherwise that call will crash
+   *
+   * @param config the system configuration
+   */
+  public static void init(final Config config) {
+    ExceptionProfiling.config = config;
+  }
+
+  /**
+   * Get a pre-configured shared instance. !!! BEFORE FIRST CALL OF THIS METHOD {@linkplain
+   * ExceptionProfiling#init(Config)} MUST HAVE BEEN INVOKED
+   *
+   * @return the shared instance
+   * @throws NullPointerException if {@linkplain ExceptionProfiling#init(Config)} has not been
+   *     called yet
+   */
+  public static ExceptionProfiling getInstance() {
+    assert config != null;
```

Blowing up in advice might not be a very good strategy... Would it be possible to gracefully skip exception handling instead?

---

IDK. The idea was that this should blow up during development or staging, etc. - since it is depending solely on calling `init(config)` in right order. If it is silently ignored there is a good chance no one will ever inspect the logs and notice that we exception profiling is not working because of this.

---

We should have integration test that verifies that exception events are being written. This assertion is likely (and hopefully) ignored by `@Advice.OnMethodExit(suppress = Throwable.class)` - so as things currently stay I do not think it is useful anyway...